### ISSUE TITLE
Object Spread can not remove property with number typed key

### ISSUE THREAD

**Issue by ele828** [babel/babel#6152]
### Input Code

```js
const { 1: _, ...rest} = { 1: 1, 2: 2};
console.log(rest); // { 1: 1, 2: 2}
```

### Babel Configuration (.babelrc, package.json, cli command)
Stage-3

### Expected Behavior

When use object spread syntax to remove a property whose key is a Number, the property should be removed correctly.

### Current Behavior
The property with Number Typed key can not be removed.

### Possible Solution

In function `_objectWithoutProperties`, avoid using `for...in` statement, because it will regard all keys as String type.

### Context

It's inconsistent with Google Chrome 60.

**Comment by nicolo-ribaudo** [comment]
Thank you for this report.
This problem has already been fixed in Babel 7 (probably by #5757)
http://babeljs.io/repl/build/4688/#?evaluate=true&lineWrap=false&presets=stage-1&babili=false&browsers=android%20%3E%202.3&builtIns=false&code_lz=G4QwTgBA3hCMBcED6AaCA6TIIF8IF4AoCE6ORWNAJkStwG5jSmBjAewDsBnNgGwFN0vNgHMAFCACUQA&debug=false&prettier=false&showSidebar=true

**Comment by jridgewell** [comment]
Yup.
